sklearn/utils/tests/test_extmath.py

```diff
@@ -148,14 +148,16 @@ def test_norm_squared_norm():
 
 def test_row_norms():
     X = np.random.RandomState(42).randn(100, 100)
-    sq_norm = (X ** 2).sum(axis=1)
+    for dtype in (np.float32, np.float64):
+        X = X.astype(dtype)
+        sq_norm = (X ** 2).sum(axis=1)
 
-    assert_array_almost_equal(sq_norm, row_norms(X, squared=True), 5)
-    assert_array_almost_equal(np.sqrt(sq_norm), row_norms(X))
+        assert_array_almost_equal(sq_norm, row_norms(X, squared=True), 4)
```

Comment:
It might be a good idea to test for the `float64` and `float32` dtype separately, with a higher precision for the float64 dtype. WDYT?